docs/_ext/autoclass_with_examples.py

```diff
@@ -100,14 +96,18 @@ def _create_hint(file_name: str, method_name: str) -> str:
         """
 
 
-def _append_hint(method_name: str, method, base_class: Any, hint: str):
+def _append_hint(method_name: str, method, parent_class: Any, hint: str):
     """
-    If method does not have the __doc__, takes it from the base method.
+    If method does not have the __doc__, takes it from the ancestor method.
     """
-    if method.__doc__ is None and method_name in base_class.__dict__:
-        method.__doc__ = getattr(base_class, method_name).__doc__ + hint
-    else:
-        method.__doc__ = (method.__doc__ or "") + hint
+    parent_method = method
+    while parent_method and parent_method.__doc__ is None:
+        parent_method = getattr(parent_class, method_name, None)
+        parent_class = parent_class.__base__
```

Comment:
I think this method taking `parent_class` as argument and renaming `parent_method = method` makes this more confusing that it needs to be. How about this function accepted a method name and related class and did the resolving internally? 

I don't see a good reason why this has to take in `base_class` as an argument.

Reply:
I didn't want to change the rest of the logic much, but okay 👍